### ISSUE TITLE
data loss if process killed

### ISSUE THREAD

**Issue by clayly** [AndersNewman/DataPoolTest#4]
Hi! Will the data be lost if I kill the process? For example like this:

```shell
kill -9 {process_id}
```

**Comment by AndersNewman** [comment]
> Hi! Will the data be lost if I kill the process? For example like this:
> 
> ```shell
> kill -9 {process_id}
> ```

Yes. The data will be lost if you kill the process. The data saved in json file in saveData() method. 
By default this method calling when program finishing(for example in "finally" block).
You can call this method after every writing operation, but this method isn't thread safe. 
Solutions:
1) Сall periodically a method saveData(). For example once a minute.
2) Make saveData() thread safe and use after every write operation.

**Comment by clayly** [comment]
Does this behavior conflict with the assignment?

**Comment by AndersNewman** [comment]
> Данные не должны теряться после остановки приложения.

Нет, данное поведение не конфликтует с условием задания. Данные не теряются, если приложение корректно заканчивает работу. 
В идеальном случае необходимо предусмотреть все непредвиденные и некорректные варианты завершения работы и реализовать подходящее решение, но в условии задания не было сказано об этом явно. 

Самое быстрое решение: обозначить метод saveData() как synchronized и вызывать его в конце метода write().
Данное решение не является производительным.

**Comment by AndersNewman** [comment]
Данное тестовое задание выполнялось мной с поправкой на ограниченное время, поэтому решение в любом случае нельзя назвать идеальным.
